packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,592 @@
+   * @memberof MgtFileList
+   */
+  @property({
+    attribute: 'load-more-file-count',
+    type: Number
+  })
+  public loadMoreFileCount: number;
+
+  private _fileListQuery: string;
+  private _fileQueries: string[];
+  private _files: MicrosoftGraph.DriveItem[];
+  private _siteId: string;
+  private _itemId: string;
+  private _driveId: string;
+  private _itemPath: string;
+  private _groupId: string;
+  private _insightType: OfficeGraphInsightString;
+  private _fileExtensions: string[];
+  private _userId: string;
+  private _renderedFileCount: number;
+  private _filesToRender: MicrosoftGraph.DriveItem[];
+
+  constructor() {
+    super();
+
+    this.showMax = 5;
+    this.loadMoreFileCount = 5;
+    this.view = ViewType.twolines;
+  }
+
+  public render() {
+    if (!this.files && this.isLoadingState) {
+      return this.renderLoading();
+    }
+
+    if (!this.files || this.files.length === 0) {
+      return this.renderNoData();
+    }
+
+    return this.renderTemplate('default', { files: this.files }) || this.renderFiles();
+  }
+
+  /**
+   * Render the loading state
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtFileList
+   */
+  protected renderLoading(): TemplateResult {
+    return this.renderTemplate('loading', null) || html``;
+  }
+
+  /**
+   * Render the state when no data is available
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtFileList
+   */
+  protected renderNoData(): TemplateResult {
+    return this.renderTemplate('no-data', null) || html``;
+  }
+
+  /**
+   * Render the list of files.
+   *
+   * @protected
+   * @param {*} files
+   * @returns {TemplateResult}
+   * @memberof mgtFileList
+   */
+  protected renderFiles(): TemplateResult {
+    return html`
+      <div id="file-list-wrapper" class="file-list-wrapper">
+        <ul id="file-list" class="file-list">
+          ${repeat(
+            this._filesToRender,
+            f => f.id,
+            f => html`
+              <li class="file-item">
+                ${this.renderFile(f)}
+              </li>
+            `
+          )}
+          ${this._renderedFileCount < this.files.length ? this.renderMoreFileButton() : null}
+        </ul>
+      </div>
+    `;
+  }
+
+  /**
+   * Render an individual file.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof mgtFileList
+   */
+  protected renderFile(file: MicrosoftGraph.DriveItem): TemplateResult {
+    const view = this.view;
+    return (
+      this.renderTemplate('file', { file }) ||
+      html`
+        <mgt-file .fileDetails=${file} .view=${view}></mgt-file>
+      `
+    );
+  }
+
+  /**
+   * Render the button when clicked will show more files.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtFileList
+   */
+  protected renderMoreFileButton(): TemplateResult {
+    const extra = this.files.length - this._renderedFileCount;
+    return html`
+        <li class="show-more" @click=${() =>
```

Comment:
Could we use a button here instead of the `li` directly?

Reply:
I am updating this to <a>